code/edge/src/androidTest/java/com/adobe/marketing/mobile/ConfigOverridesFunctionalTests.kt

```diff
@@ -431,15 +376,21 @@ class ConfigOverridesFunctionalTests {
         assertEquals(1, resultRequests.size.toLong())
 
         // Assert that default datastream ID value is in the URL query param for configId
-        assertEquals(CONFIG_ID, resultRequests[0]!!.queryParam("configId"))
+        assertEquals(CONFIG_ID, TestableNetworkRequest(resultRequests[0]).queryParam("configId"))
 
-        val payloadJSONData = getPayloadJson(resultRequests[0])
-        val eventJSONData = payloadJSONData.getJSONArray("events").getJSONObject(0)
-        val metaJSONData = payloadJSONData.getJSONObject("meta")
-
-        JSONAsserts.assertExactMatch(expectedXdmDataMap, eventJSONData)
-        // verify that sdkConfig is not present in the meta data
-        assertNull(metaJSONData.opt("sdkConfig"))
+        val expected = """
+        {
+          "events": [
+            {
+              "xdm": {
+                "testXdmKey": "testXdmValue"
+              }
+            }
+          ]
+        }
+        """
+        // Verify that sdkConfig is not present in the meta data
+        assertExactMatch(expected, getPayloadJson(resultRequests[0]), KeyMustBeAbsent("meta.sdkConfig"))
```

Comment:
If we are converting to TestableNetworkRequest for queryParam, we can move the getPayloadJson to that class as well.